File: src/test/java/com/testexercise/ebayframework/BasePage.java
package com.testexercise.ebayframework;

import org.openqa.selenium.*;

public class BasePage
{
    protected final AutomationFramework af;
    
    public BasePage(AutomationFramework af)
    {
        this.af = af;
    }
    
    final String searchTextInputCss = "form#gh-f input[type='text']";
    final String searchSubmitCss = "form#gh-f input[type='submit']";

    public WebElement searchBox()
    {
        return af.driver().findElement(By.cssSelector(searchTextInputCss));
    }
    
    public WebElement searchSubmit()
    {
        return af.driver().findElement(By.cssSelector(searchSubmitCss));
    }
    
    public void enterSearchQuery(String searchQuery)
    {
        searchBox().clear();
        searchBox().sendKeys(searchQuery);
    }
    
    public void submitSearch()
    {
        searchSubmit().click();
    }
    
    public String getPageUri()
    {
        return af.driver().getCurrentUrl();
    }
}

File: net/TestExerciseEbay/EbayPOMFramework/BasePage.cs
using System;
using OpenQA.Selenium;

namespace TestExercise.EbayPOMFramework
{ 
    public class BasePage
    {
        protected readonly AutomationFramework af;

        public BasePage(AutomationFramework af)
        {
            this.af = af;
        }

        const string searchTextInputCss = "form#gh-f input[type='text']";
        const string searchSubmitCss = "form#gh-f input[type='submit']";

        public IWebElement SearchBox()
        {
            return af.Driver.FindElement(By.CssSelector(searchTextInputCss));
        }

        public IWebElement SearchSubmit()
        {
            return af.Driver.FindElement(By.CssSelector(searchSubmitCss));
        }

        public void EnterSearchQuery(String searchQuery)
        {
            SearchBox().Clear();
            SearchBox().SendKeys(searchQuery);
        }

        public void SubmitSearch()
        {
            SearchSubmit().Click();
        }

        public string PageUri => af.Driver.Url;
    }
}